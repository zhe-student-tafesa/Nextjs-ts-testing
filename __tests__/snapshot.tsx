import { render } from '@testing-library/react'  // The purpose of the "render" function is to render a React component into a virtual DOM that can be tested.
import HomePage from '../src/app/page'
 
it('renders homepage unchanged', () => {
  const { container } = render(<HomePage />)
// Jest saves the rendered DOM structure of the container as a snapshot file (usually in the __snapshots__ directory).
// If the rendered DOM changes during subsequent tests, Jest will alert you to the snapshot mismatch, helping you identify unexpected changes in your component.
// Use case: Ensuring that the UI output remains consistent with the previous snapshot.
  expect(container).toMatchSnapshot()
})