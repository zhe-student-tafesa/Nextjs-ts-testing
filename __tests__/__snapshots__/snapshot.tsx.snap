// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`renders homepage unchanged 1`] = `
<div>
  <div>
    <h1>
      Home
    </h1>
    <a
      href="/about"
    >
      About
    </a>
  </div>
</div>
`;
